starknet-rpc-test/get_transaction_by_blockid_and_index.rs

```diff
@@ -0,0 +1,139 @@
+extern crate starknet_rpc_test;
+
+use assert_matches::assert_matches;
+use rstest::rstest;
+use starknet_accounts::SingleOwnerAccount;
+use starknet_core::chain_id;
+use starknet_core::types::{
+    BlockId, BlockTag, DeclareTransaction, InvokeTransaction, InvokeTransactionV1, MaybePendingBlockWithTxs,
+    StarknetError, Transaction,
+};
+use starknet_ff::FieldElement;
+use starknet_providers::ProviderError::StarknetError as StarknetProviderError;
+use starknet_providers::{MaybeUnknownErrorCode, Provider, StarknetErrorWithMessage};
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, MINT_AMOUNT, SIGNER_PRIVATE, TEST_CONTRACT_CLASS_HASH};
+use starknet_rpc_test::fixtures::madara;
+use starknet_rpc_test::utils::AccountActions;
+use starknet_rpc_test::{MadaraClient, Transaction as TransactionEnum};
+use starknet_signers::{LocalWallet, SigningKey};
+
+#[rstest]
+#[tokio::test]
+async fn fail_non_existing_block(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    assert_matches!(
+        rpc
+        .get_transaction_by_block_id_and_index(
+            BlockId::Number(1),
+            0
+        )
+        .await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code:
+MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::BlockNotFound     );
```

Comment:
Because `StarknetError` is an enum I think you can even do this:

`Err(StarknetProviderError(StarknetErrorWithMessage { code:
MaybeUnknownErrorCode::Known(StarknetError::BlockNotFound), .. }))`

Without the `if` part